src/rail_gui.cpp

```diff
@@ -891,6 +895,30 @@ struct BuildRailStationWindow : public PickerWindowBase {
 	Scrollbar *vscroll;   ///< Vertical scrollbar of the new station list.
 	Scrollbar *vscroll2;  ///< Vertical scrollbar of the matrix with new stations.
 
+	typedef GUIList<StationClassID, StringFilter &> GUIStationClassList; ///< Type definition for the list to hold available station classes.
+
+	static const uint EDITBOX_MAX_SIZE = 16; ///< The maximum number of characters for the filter edit box.
+
+	static Listing   last_sorting;           ///< Default sorting of #GUIStationClassList.
+	static Filtering last_filtering;         ///< Default filtering of #GUIStationClassList.
+	static GUIStationClassList::SortFunction * const sorter_funcs[];   ///< Sort functions of the #GUIStationClassList.
+	static GUIStationClassList::FilterFunction * const filter_funcs[]; ///< Filter functions of the #GUIStationClassList.
+	GUIStationClassList station_classes;     ///< Available station classes.
+	StringFilter string_filter;              ///< Filter for available station classes.
+	QueryString filter_editbox;              ///< Filter editbox.
+
+	/** Scroll #WID_BRAS_NEWST_SCROLL so that the selected station class is visible. */
+	void EnsureSelectedStationClassIsVisible()
+	{
+		uint pos = 0;
+		for (auto station_class : this->station_classes) {
+			if (station_class == _railstation.station_class) break;
+			pos++;
+		}
```

Comment:
```suggestion
		uint pos = std::find(this->station_classes.begin(), this->station_classes.end(), _railstation.station_class) - this->station_classes.begin();
```
Though I'm undecided whether this is actually better. Might need some "was class actually found" checking